GitCommands/AppTitleGenerator.cs

```diff
@@ -46,10 +50,17 @@ public string Generate(string workingDir = null, bool isValidWorkingDir = false,
             var description = _description.Get(workingDir);
 
 #if DEBUG
-            return $"{description} ({branchName}) - Git Extensions [DEBUG]";
+            return $"{description} ({branchName}) - Git Extensions [{BuildBranch ?? "DEBUG"}]";
 #else
             return $"{description} ({branchName}) - Git Extensions";
+#endif
+        }
+
+        public static void Initialise(string buildBranch)
```

Comment:
I proposed the property in order to replace this method.

Reply:
If this is changed, the #if DEBUG need to be handled in Program.cs too - is that requested?

Reply:
I'd make the property independent from `DEBUG`. Then it will just not be used. @RussKie?

Reply:
I'd like to ponder on this a little. I suspect we can do it slightly differently, but I need some time to investigate.

Reply:
Fine, I just seen some needs to track my private builds

Reply:
> Any reviews, to merge or dismiss this PR?

The private property `BuildBranch` is pointless IMO. (I don't know which is a better English word if this was too harsh.)
Either a non-`DEBUG` public propery `BuildBranch` or `Initialise`. I prefer the first for readability.

Reply:
@RussKie, what do you think about non-`DEBUG` public propery `BuildBranch` vs. `Initialise`?

Reply:
I personally don't like `public static` properties. I don't like `public static` method either, but in this case it feels like a lesser evil. And we already have another object with the same behaviour.

I also entertained an idea of implementing a debug-only version of `IAppTitleGenerator` and inject it, but it felt like more work for little gain. Besides we haven't fleshed out the DI story (MEF could be the way to go, but I haven't figured how it fits with the testing story).